microsoft/nni
# DNGO tuner

DNGO works as a NAS tuner
Scalable Bayesian Optimization Using Deep Neural Networks(DNGO):  https://arxiv.org/pdf/1502.05700.pdf 
Using DNGO model in https://github.com/automl/pybnn

how to test:
     1. make sure codeDir in config.yml is correct
     2. nnictl create -c examples/tuners/dngo_tuner/config.yml 
search space:
      support both numerical and string type(convert string to index as its value so that DNGO can get) 
      support choice/randint/uniform/quniform/loguniform/qloguniform(by reusing nni.parameter_expressions)
